logic/src/commonMain/kotlin/com/wire/kalium/logic/data/asset/AssetMessage.kt

```diff
@@ -30,7 +29,69 @@ data class AssetMessage(
     val assetId: String,
     val width: Int,
     val height: Int,
-    val downloadStatus: Message.DownloadStatus,
     val assetPath: Path?,
     val isSelfAsset: Boolean
 )
+
+enum class AssetTransferStatus {
+    /**
+     * There was no attempt done to transfer the asset's data to/from remote (server) storage.
+     */
+    NOT_PROCESSED,
+
+    /**
+     * The asset is currently being uploaded to remote storage.
+     */
+    UPLOAD_IN_PROGRESS,
+
+    /**
+     * The asset is currently being downloaded from remote storage.
+     */
+    DOWNLOAD_IN_PROGRESS,
+
+    /**
+     * The asset was successfully uploaded and saved in the internal storage.
+     */
+    UPLOADED,
+
+    /**
+     * The asset has been successfully downloaded and saved in the internal storage,
+     * that should be only readable by this client. This state is used for downloaded assets saved internally.
+     */
+    SAVED_INTERNALLY,
+
+    /**
+     * The asset was downloaded and saved in an external storage, readable by other software on the machine.
+     * e.g.: Asset was saved in Downloads, Desktop, or other user-chosen directory.
+     */
+    SAVED_EXTERNALLY,
+
+    /**
+     * The last attempt at uploading this asset's data failed.
+     */
+    FAILED_UPLOAD,
+
+    /**
+     * The last attempt at downloading this asset's data failed.
+     */
+    FAILED_DOWNLOAD,
+
+    /**
+     * Asset was not found on the server.
+     */
+    NOT_FOUND
+}
+
+fun AssetTransferStatus.isInProgress(): Boolean {
+    return this == AssetTransferStatus.UPLOAD_IN_PROGRESS || this == AssetTransferStatus.DOWNLOAD_IN_PROGRESS
+}
+
+fun AssetTransferStatus.isFailed(): Boolean {
+    return this == AssetTransferStatus.FAILED_DOWNLOAD || this == AssetTransferStatus.FAILED_UPLOAD
+}
+
+fun AssetTransferStatus?.isSaved(): Boolean {
+    return this == AssetTransferStatus.SAVED_EXTERNALLY
+            || this == AssetTransferStatus.SAVED_INTERNALLY
+            || this == AssetTransferStatus.UPLOADED
```

Comment:
It is in the documentation

```   
    /**
     * The asset was successfully uploaded and saved in the internal storage.
     */
    UPLOADED,
```

Reply:
It still doesn't fully tell because it can be uploaded by you or by someone else - I think it wouldn't hurt to explain it a bit more, like "Upload states up to this one only apply to assets being uploaded by the current client. Assets uploaded by other clients and received in a message start with `NOT_DOWNLOADED` state."  😉